pipeline/scripts/setup-emulator.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+emulator_config="system-images;android-$ANDROID_API_LEVEL;google_apis;x86_64"
+emulator_name="emulator_api_$ANDROID_API_LEVEL"
+
+# Install AVD files
+echo "y" | $ANDROID_HOME/tools/bin/sdkmanager --install "$emulator_config"
+# Create emulator
+echo "no" | $ANDROID_HOME/tools/bin/avdmanager create avd -n "$emulator_name" -k "$emulator_config" --force
+
+$ANDROID_HOME/emulator/emulator -list-avds
+
+echo "Starting emulator"
+
+# Start emulator in background
+nohup $ANDROID_HOME/emulator/emulator -avd "$emulator_name" -no-snapshot -no-boot-anim > /dev/null 2>&1 &
+$ANDROID_HOME/platform-tools/adb wait-for-device shell 'while [[ -z $(getprop sys.boot_completed | tr -d '\r') ]]; do sleep 1; done; input keyevent 82'
```

Comment:
I had some issues with the device not actually being ready despite this line, so I added some somewhat arbitrary delays below. I would love to replace the delays with a more intelligent approach if anyone has any suggestions

Reply:
My initial thought is to try ending the script with a similar sort of loop, but based on whether the results of some minimal `adb shell content` command contains the string "Could not find provider".

```bash
$ANDROID_HOME/platform-tools/adb shell 'while [[ $(content gettype --uri content://com.microsoft.accessibilityinsightsforandroidservice/config 2>&1 | grep "Could not find provider") ]]; do sleep 1; done'
```

Reply:
Ideally, I'd like to avoid any interaction with the service in this script so that any errors show up in the "testing" build tasks. I was able to significantly reduce the delays here from what I initially had, so the current approach feels a bit better to me. 

Open to adding your suggestion if you think it's a better approach though!